examples/miscellaneous/plot_outlier_detection_bench.py

```diff
@@ -0,0 +1,157 @@
+"""
+==========================================
+Evaluation of outlier detection estimators
+==========================================
+
+This example benchmarks outlier detection algorithms using ROC curves
+on classical anomaly detection datasets. The algorithm performance
+is assessed in an outlier detection context:
+
+1. The algorithms are trained on the whole dataset which is assumed to
+contain outliers.
+
+2. The ROC curve is computed on the same dataset using the knowledge
+of the labels.
+
+
+Interpreting the ROC plot
+-------------------------
+The algorithm performance relates to how good the true positive rate (TPR)
+is at low value of the false positive rate (FPR). The better algorithms
+have the curve on the top-left of the plot and the area under curve (AUC)
+close to 1. The diagonal dashed line represents a random classification
+of outliers and inliers.
+"""
+
+from time import time
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.neighbors import LocalOutlierFactor
+from sklearn.ensemble import IsolationForest
+from sklearn.metrics import roc_curve, auc
+from sklearn.datasets import fetch_kddcup99, fetch_covtype, fetch_openml
+from sklearn.preprocessing import LabelBinarizer
+
+print(__doc__)
+
+rng = np.random.RandomState(42)
+
+# datasets
+datasets = ["http", "smtp", "SA", "SF", "forestcover",
+            "glass", "wdbc", "cardiotocography"]
+
+# outlier detection models
+models = [
+    ("LOF", LocalOutlierFactor(n_neighbors=20, contamination="auto")),
+    ("IForest", IsolationForest(random_state=rng, contamination="auto")),
+]
+
+rows = math.ceil(len(datasets)/2)
+plt.figure(figsize=(10, rows * 3))
+for dataset_idx, dataset_name in enumerate(datasets):
+    plt.subplot(rows, 2, dataset_idx + 1)
+
+    # loading and vectorization
+    print(f"Loading {dataset_name} data")
+    if dataset_name in ["http", "smtp", "SA", "SF"]:
+        dataset = fetch_kddcup99(
+            subset=dataset_name, percent10=True, random_state=rng
+        )
+        X = dataset.data
+        y = dataset.target
+
+    if dataset_name == "forestcover":
```

Comment:
please put this after all the kdd cup datasets so that kdd cup dataset preprocessing follows fetch_kddcup99

Reply:
Can you clarify what do you mean in this comment? Do I add `fetch_kddcup99(subset=None)` or do I put 
```
x = dataset.data
y = dataset.target
```
below all `if` in the loading data section?

Reply:
Or should we merge loading and vectorizing to be one section?

Reply:
Now I got it. My change is in the next commit

Reply:
sorry this was not very clear